### PR TITLE
[optimize] Use gzip to compress data #64

### LINKED ISSUE #64: [optimize] Use gzip to compress data
We can reduce the use of memory of more than 70% thanks to gzip.

We should just do the following:

- a basic gzip is provided by busybox, this should be enough, I am not sure whether we should specify this in the dependencies or not, we should double check this
- gzip files after saving them, eg: `gzip $path`, this command also removes the original file
- decompress files before sending the data, eg: `gzip -d $path`

### PR DESCRIPTION
**Pull request by devkapilbansal**
Closes #64

### CODE REVIEW

File: openwrt-openwisp-monitoring/Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ define Package/openwisp-monitoring/default
 	CATEGORY:=Administration
 	SECTION:=admin
 	SUBMENU:=openwisp
-	DEPENDS:=+netjson-monitoring $(3)
+	DEPENDS:=+netjson-monitoring +gzip $(3)
```
Comment (nemesisdesign) [inline openwrt-openwisp-monitoring/Makefile:22]: wait, we don't need the full gzip, we should be able to use the gzip utility provided by busy box, which is shipped by default in OpenWrt:

```
gzip --help
BusyBox v1.33.1 (2021-07-31 21:08:50 UTC) multi-call binary.

Usage: gzip [-cfkdt] [FILE]...

Compress FILEs (or stdin)

	-d	Decompress
	-t	Test file integrity
	-c	Write to stdout
	-f	Force
	-k	Keep input files
``` 

We are space constrained on these devices, we have to be extra careful before adding a new dependency.

File: openwrt-openwisp-monitoring/files/monitoring.agent [hunk 2]
```diff
@@ -188,8 +186,11 @@ send_data() {
 					fi
 					break
 				fi
+				#decompress file
```
Comment (nemesisdesign) [inline openwrt-openwisp-monitoring/files/monitoring.agent:189]: remember to leave a space after the `#`

**Review by nemesisdesign — COMMENTED** [review]
Good progress @devkapilbansal, please see my comments below.

File: openwrt-openwisp-monitoring/Makefile (hunk 1 above)
Comment (devkapilbansal) [reply openwrt-openwisp-monitoring/Makefile:22]: If `gzip` is already available, this shouldn't take any extra space. However, I removed it as I am not sure if the gzip package is heavier than the one provided by busy box

File: openwrt-openwisp-monitoring/files/monitoring.agent [hunk 3]
```diff
@@ -213,6 +218,8 @@ send_data() {
 					sleep "$timeout"
 				fi
 			done
+			# compress file again
+			gzip "$filename"
```
Comment (nemesisdesign) [inline openwrt-openwisp-monitoring/files/monitoring.agent:222]: is it really needed?

Comment (devkapilbansal) [reply openwrt-openwisp-monitoring/files/monitoring.agent:222]: I think not now :thinking: 

I added it when we were checking for `.gz` files only. Removing it

**Review by nemesisdesign — APPROVED** [review]
Thanks! :+1:

**Pull request merged**